packages/cicero-server/app.js

```diff
@@ -35,47 +48,74 @@ app.use(bodyParser.json());
  *
  * Response
  * ----------
  * JSON formated response object
  *
  */
-app.post('/trigger/:template', async function (req, httpResponse, next) {
+app.post('/trigger/:type/:instanceName', async function (req, httpResponse, next) {
     try {
-        const clause = await initTemplateInstance(req);
+        const instance = req.params.type === 'clause' ? await initTemplateInstance(req) : req.params.type === 'contract' ? await loadInstance(req) : null;
+        if(!instance) {
+            throw new Error('The type specified is wrong. Please use either \'clause\' or \'contract\' type');
+        }
+        if(req.params.type === 'clause') {
+            instance.setData(req.body.data);
+        }
+        if(req.params.type === 'contract' && !req.body.partyName) {
+            throw new Error('Please enter the name of the party that is triggering the contract Instance');
+        }
+        if(req.params.type === 'clause' && !Object.prototype.hasOwnProperty.call(req.body,'data')) {
+            throw new Error('Missing data in /trigger body');
+        }
 
         const engine = new Engine();
         let result;
         if(Object.keys(req.body).length === 3 &&
            Object.prototype.hasOwnProperty.call(req.body,'request') &&
-           Object.prototype.hasOwnProperty.call(req.body,'state') &&
-           Object.prototype.hasOwnProperty.call(req.body,'data')) {
-            clause.setData(req.body.data);
-            result = await engine.trigger(clause, req.body.request, req.body.state);
+           Object.prototype.hasOwnProperty.call(req.body,'state')) {
+            result = await engine.trigger(instance, req.body.request, req.body.state);
         } else if(Object.keys(req.body).length === 2 &&
-           Object.prototype.hasOwnProperty.call(req.body,'request') &&
-           Object.prototype.hasOwnProperty.call(req.body,'data')) {
+           Object.prototype.hasOwnProperty.call(req.body,'request')) {
             const state = { '$class' : 'org.accordproject.runtime.State' };
-            clause.setData(req.body.data);
-            result = await engine.trigger(clause, req.body.request, state);
+            result = await engine.trigger(instance, req.body.request, state);
             delete result.state;
         } else {
-            throw new Error('Missing request, state or data in /trigger body');
+            throw new Error('Missing request or state in /trigger body');
+        }
+        if(req.params.type === 'contract') {
+            //Add state
+            Util.addHistory(
+                instance,
+                req.body.partyName,
+                'trigger',
+                result,
+                'Execution'
+            );
+            const archive = await instance.toArchive('ergo');
+            const file = !req.body.output ? `${process.env.CICERO_CONTRACTS}/${req.params.instanceName}.slc` : `${process.env.CICERO_CONTRACTS}/${req.body.output}.slc`;
+            fs.writeFileSync(file, archive);
```

Comment:
Assumes we have write access to the mounted file system - which may not be true in Docker or Cloud.

Reply:
Agreed. Let's go with this for the time being and fix in subsequent PR(s). TBD.